raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -163,12 +163,14 @@
     private final QuorumState quorum;
     private final Random random;
     private final ConnectionCache connections;
-    private final FuturePurgatory<Void> purgatory;
+    private final FuturePurgatory<Void> fetchPurgatory;
+    private final FuturePurgatory<OffsetAndEpoch> appendPurgatory;  // note we would only use this purgatory to expire requests,
```

Comment:
It's a bit annoying that we need two collections to track uncommitted appends. To converge the two, it seems we need the `tryComplete` functionality from purgatory?

Reply:
Yup, confirmed.